database.go

```diff
@@ -49,34 +51,60 @@ func migrationUp() error {
 }
 
 /* Verifies if there is already a book with the name of "newBook" in the database. If yes, returns it. */
-func sameNameOnDB(newBook Book) (unique bool) {
+func sameNameOnDB(newBook Book) (unique bool, unexpected error) {
 	sqlStatement := `SELECT true FROM bookstable WHERE name=$1;`
-	foundRow := dbObject.QueryRow(sqlStatement, newBook.Name)
+	foundRow := dbObjectGlobal.QueryRow(sqlStatement, newBook.Name)
 	var bookAlreadyExists bool
-	switch err := foundRow.Scan(&bookAlreadyExists); err {
+	err := foundRow.Scan(&bookAlreadyExists)
+	switch err {
 	case sql.ErrNoRows:
-		return true
+		return true, nil
 	case nil:
-		return false
+		return false, nil
 	default:
-		panic(err)
+		return false, fmt.Errorf("verifying same name on db: %w", err)
 	}
 }
 
+var errBookNotFound = errors.New("book not found")
+
+/* Search a book in database based on ID and returns it if succeed. */
+func searchById(id uuid.UUID) (Book, error) {
+	sqlStatement := `SELECT id, name, price, inventory FROM bookstable WHERE id=$1;`
+	foundRow := dbObjectGlobal.QueryRow(sqlStatement, id)
+	var bookToReturn Book
+	err := foundRow.Scan(&bookToReturn.ID, &bookToReturn.Name, &bookToReturn.Price, &bookToReturn.Inventory)
+	if err != nil {
+		switch err {
+		case sql.ErrNoRows:
+			{
+				return Book{}, fmt.Errorf("searching by ID: %w", errBookNotFound)
+			}
```

Comment:
I am sorry, I let this slip. The brackets are not needed here.
```suggestion
			return Book{}, fmt.Errorf("searching by ID: %w", errBookNotFound)